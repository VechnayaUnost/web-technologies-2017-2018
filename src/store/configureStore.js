import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware, { END } from 'redux-saga';

import rootReducer from '../reducers';
import rootSaga from '../sagas';

export default (initialState) => {
    const sagaMiddleware = createSagaMiddleware();
    const store = createStore(rootReducer, initialState, applyMiddleware(sagaMiddleware));

    sagaMiddleware.run(rootSaga);
    store.runSaga = () => sagaMiddleware.run(rootSaga);
    store.close = () => store.dispatch(END);

    return store;
};