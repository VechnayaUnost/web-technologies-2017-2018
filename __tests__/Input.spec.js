import React from 'react';
import Input from '../src/components/Input';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

describe('Input component test', () => {
    const mockStore = configureStore([thunk]);
    const initialState = {}

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('Input component rendered succesfully', () => {
        const input = mount(
            <Input store={store} />
        );
        expect(mountToJson(input)).toMatchSnapshot();
    });

    it('Input component event triggers', () => {
        const props = {
            fetchUserRequest: jest.fn()
        }
        const input = mount(
            <Input store={store} {...props}/>
        );
        input.find('input').simulate('change', { target: { value: 'meow' } });
        input.find('input').simulate('keyDown', { key: 'Enter', keyCode: 13, which: 13, shiftKey: false });
        input.setState({text: 'meow'});
        input.find('button').simulate('click');
        input.instance().props.fetchUserRequest(input.find('button').simulate('click'));
        expect(props.fetchUserRequest).toHaveBeenCalled();
    });
});