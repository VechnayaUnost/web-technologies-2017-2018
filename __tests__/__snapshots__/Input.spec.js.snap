// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component test Input component rendered succesfully 1`] = `
<Connect(Input)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Input
    fetchUserRequest={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <div
      className="inputs"
    >
      <input
        id="inputText"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Username"
        type="text"
      />
      <button
        onClick={[Function]}
      >
        Search
      </button>
    </div>
  </Input>
</Connect(Input)>
`;
